WordPress/gutenberg
# Patterns: split up the block editor inserter patterns tab into separate component files

## What?

Refactors the block editor inserter patterns tab in to separate component files and renames some components for clarity.

## Why?

The `packages/block-editor/src/components/inserter/block-patterns-tab.js` file has grown quite large over time and contains a mix of hooks, components and util files which makes it more difficult to work on. Also the component naming is a little confusing with a dialog component loading a panel component.

## How?

Breaks all the individual parts of the patterns inserter tab out into separate files in a subfolder. Also renames the dialog and preview components. The pattern previews panel is more of a panel than a dialog so renames the relevant components to better indicate the relationship.

## Testing Instructions

- In the block editor test all aspects of the inserter patterns tab, check categories display as expected, preview panel works, filtering in preview panel, paging for larger categories, patterns insert correctly when selected, etc.
- Check that the explorer panel also works as expected still
